docarray/index/backends/in_memory.py

```diff
@@ -293,13 +291,53 @@ def execute_query(self, query: List[Tuple[str, Dict]], *args, **kwargs) -> Any:
             raise ValueError(
                 f'args and kwargs not supported for `execute_query` on {type(self)}'
             )
-        find_res = _execute_find_and_filter_query(
-            doc_index=self,
-            query=query,
-            reverse_order=True,
-        )
+        find_res = self._hybrid_search(query)
         return find_res
 
+    def _hybrid_search(self, query: List[Tuple[str, Dict]]) -> FindResult:
```

Comment:
hybrid search is not that. This is find and filter, let's change the name of this method for clarity

Reply:
no, hybrid search is about mixing bm25 scores with embedding scores. Not about filtering + search

Reply:
I think hybrid search means combining more than one search methods. bm25 + vector search is just one case.

Even we use it like that 😄 
https://docs.docarray.org/user_guide/storing/docindex/#hybrid-search-through-the-query-builder